### PR TITLE
adds net-restrict to cli options

### LINKED ISSUE #1625: Document subnet-based peer permissions
Document new feature for managing peer permissions based on IP subnet configurations in the networking layer

**Related to:** https://github.com/hyperledger/besu/pull/7168

### PR DESCRIPTION
**Pull request by m4sterbunny**
## Description
Adds net-restrict to CLI options (supports Comma-separated array of allowed IP subnets). 

### Issue(s) fixed

Fixes #1625 

### Preview

-

### CODE REVIEW

File: docs/public-networks/reference/cli/options.md [hunk 1]
```diff
@@ -2596,6 +2596,46 @@ You must specify `DOCKER` when using the [Besu Docker image](../../get-started/i
...
 :::
 
+### `net-restrict`
+
+<Tabs>
+
+<TabItem value="Syntax" label="Syntax" default>
+
+```bash
+--net-restrict=[<String>,..]
```
Comment (alexandratran) [inline docs/public-networks/reference/cli/options.md:2606]:
```suggestion
--net-restrict=<subnet>[,<subnet>,...]
```

File: docs/public-networks/reference/cli/options.md [hunk 2]
```diff
@@ -2596,6 +2596,46 @@ You must specify `DOCKER` when using the [Besu Docker image](../../get-started/i
...
+```bash
+--net-restrict=192.168.1.0/24,10.0.0.0/8
+```
+
+</TabItem>
+
+<TabItem value="Environment variable" label="Environment variable">
+
+```bash
+BESU_NET-RESTRICT=192.168.1.0/24,10.0.0.0/8
```
Comment (macfarla) [inline docs/public-networks/reference/cli/options.md:2622]:
```suggestion
BESU_NET_RESTRICT=192.168.1.0/24,10.0.0.0/8
```

**Comment by m4sterbunny** [comment]
switching to draft to add info to https://besu.hyperledger.org/development/public-networks/how-to/connect/manage-peers

File: docs/public-networks/how-to/connect/manage-peers.md [hunk 3]
```diff
@@ -12,10 +12,12 @@ tags:
...
-The frequency of discovery isn't configurable, but you can
-[limit remote connections](#limit-remote-connections) in public networks and
-[randomly prioritize connections](../../reference/cli/options.md#random-peer-priority-enabled) in
-small, stable networks.
+The frequency of discovery isn't configurable, but you can:
+- [Limit remote connections](#limit-remote-connections) in public networks
+- [Allowlist peers by IP subnet](#allowlist-peers) to create a private network of peers across public 
+networks
+- [Randomly prioritize connections](../../reference/cli/options.md#random-peer-priority-enabled) in
+small, stable networks
```
Comment (alexandratran) [inline docs/public-networks/how-to/connect/manage-peers.md:20]:
```suggestion
The frequency of discovery isn't configurable, but you can:

- [Limit remote connections](#limit-remote-connections) in public networks.
- [Allowlist peers by IP subnet](#allowlist-peers) to create a private network of peers across public 
networks.
- [Randomly prioritize connections](../../reference/cli/options.md#random-peer-priority-enabled) in
small, stable networks.
```

File: docs/public-networks/how-to/connect/manage-peers.md [hunk 4]
```diff
@@ -70,6 +72,10 @@ You can limit peers to reduce the bandwidth, CPU time, and disk access Besu uses
 
 To reduce the maximum number of peers, use the [`--max-peers`](../../reference/cli/options.md#max-peers) option. The default is 25.
 
+## Allowlist peers
+
+You can can define specific IP subnets permitted to interact with the node using the [`--net-restrict`](../../reference/cli/options.md#net-restrict) configuration. This restricts access to only those peers whose IP addresses fall within the allowed subnets. This is particularly useful if you maintain a set of nodes and want to restrict which of those may connect to external nodes.
```
Comment (alexandratran) [inline docs/public-networks/how-to/connect/manage-peers.md:77]:
```suggestion
You can can define specific IP subnets permitted to interact with the node using the [`--net-restrict`](../../reference/cli/options.md#net-restrict) configuration. This restricts access to only those peers whose IP addresses fall within the allowed subnets. This is useful if you maintain a set of nodes and want to restrict which of those can connect to external nodes.
```

File: docs/public-networks/reference/cli/options.md [hunk 5]
```diff
@@ -2596,6 +2596,46 @@ You must specify `DOCKER` when using the [Besu Docker image](../../get-started/i
...
+
+```bash
+net-restrict=["192.168.1.0/24","10.0.0.0/8"]
+```
+
+</TabItem>
+
+</Tabs>
+
+Comma-separated array of allowed IP subnets. Enables node operators to nominate peers by their IP subnets. Default is null: no subnet-based peer permission restrictions will be applied.
```
Comment (alexandratran) [inline docs/public-networks/reference/cli/options.md:2637]:
```suggestion
A comma-separated list of allowed IP subnets.
Peers whose IP addresses fall within the specific subnets are granted permission to interact with the node. If not specified, no subnet-based peer permissions are applied.
```

Reword suggestion for consistency with other options and a bit more clarity.

**Review by alexandratran — APPROVED** [review]
LGTM once suggestions are addressed. This could still use a technical approval.

**Pull request merged**